content/docs/ai/ai-checkpoints.md

```diff
@@ -0,0 +1,356 @@
+---
+title: Using Snapshots as checkpoints
+subtitle: Implementing AI agent checkpoints with Neon snapshots
+enableTableOfContents: true
+updatedOn: '2025-08-12T14:09:12.290Z'
+---
+
+<ComingSoon/>
+
+## Overview
+
+This guide is for platforms building AI agents and code generation tools (like Replit Agent, v0, Lovable, Anything). It explains how to implement database checkpointing using Neon's Snapshot API an active branch pattern, which maintains a persistent root branch for production while using snapshots for checkpoints and temporary branches for previews. See a working implementation in the [demonstration repository](https://github.com/neondatabase-labs/snapshots-as-checkpoints-demo).
```

Comment:
```suggestion
This guide is for platforms building AI agents and code generation tools. It explains how to implement database checkpointing using Neon's Snapshot API an active branch pattern, which maintains a persistent root branch for production while using snapshots for checkpoints and temporary branches for previews. See a working implementation in the [demonstration repository](https://github.com/neondatabase-labs/snapshots-as-checkpoints-demo).
```

Reply:
Andre's repo is such a great complement to this guide. I wonder of we can make our call out to it more visible, maybe include a description of the app here.
Tying the docs closely to that example repo as much as possible would be great for users/readers.

Reply:
Good idea, now a demo section is near the top, and it is referenced in multiple places.